app/models/user_role.rb

```diff
@@ -11,6 +11,7 @@ class UserRole < ApplicationRecord
   delegate :group_type, to: :group
 
   scope :active, -> { where(end_date: nil).or(where.not(end_date: ..Date.today)) }
```

Comment:
Crazy Rails magic: (read `inactive.invert_where` as "where NOT inactive")

```suggestion
  scope :active, -> { where(end_date: nil).or(inactive.invert_where) }
```